lib/dir_helpers.ml

```diff
@@ -2,12 +2,16 @@
  * Copyright (c) 2024 Puneeth Chaganti <punchagan@muse-amuse.in>, Shon Feder <shon.feder@gmail.com>, Tarides <contact@tarides.com>
  *)
 
+(*Generate random temporary file names. **)
+let temp_file_name dir prefix suffix =
+  let prng = Random.State.make_self_init () in
+  let random_suffix = string_of_int (Random.State.int prng 0x10000000) in
+  Filename.concat dir (prefix ^ random_suffix ^ suffix)
+
+(*Create a temporary directory with the given prefix. **)
```

Comment:
It may be worth bringing [Bos](https://erratique.ch/software/bos), which would deal with a lot of the stuff we are having to reimplement, while being a pretty lightweight dependency.

Reply:
Thanks for your suggestion. I'll look into it!